File: src/main/app/src/app/core/websocket/model/screen-update-event.ts
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import {ObjectType} from './object-type';
import {Operatie} from './operatie';

export class ScreenUpdateEvent {
    timestamp: number;
    operatie: Operatie;
    objectType: ObjectType;
    objectId: string;
    key: string;

    constructor(operatie: Operatie, objectType: ObjectType, objectId: string, timestamp?: number) {
        this.timestamp = timestamp;
        this.operatie = operatie;
        this.objectType = objectType;
        this.objectId = objectId;
        this.key = this.operatie + ';' + this.objectType + ';' + this.objectId;
    }
}

File: src/main/java/net/atos/zac/websocket/event/ScreenUpdateEvent.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.websocket.event;

import javax.validation.constraints.NotNull;

import net.atos.zac.event.AbstractUpdateEvent;
import net.atos.zac.event.OpcodeEnum;

public class ScreenUpdateEvent extends AbstractUpdateEvent<ScreenObjectTypeEnum, String> {

    private static final long serialVersionUID = -740125186878024703L;

    @NotNull
    private ScreenObjectTypeEnum objectType;

    public ScreenUpdateEvent() {
        super();
    }

    public ScreenUpdateEvent(final OpcodeEnum opcode, final ScreenObjectTypeEnum objectType, final String objectId) {
        super(opcode, objectId);
        this.objectType = objectType;
    }

    @Override
    public ScreenObjectTypeEnum getObjectType() {
        return objectType;
    }
}
